packages/invoices/src/OutboundShipment/DetailView/reducer.ts

```diff
@@ -104,20 +104,31 @@ export const reducer = (
 
           Object.keys(draft).forEach(key => {
             // TODO: Sometimes we want to keep the user entered values?
+            if (key === 'lines') return;
             draft[key] = data[key];
           });
 
-          draft.lines = draft.lines?.map(item => createLine(item, dispatch));
+          draft.lines = data.lines?.map(serverLine => {
+            const draftLine = draft.lines.find(
+              line => line.id === serverLine.id
+            );
+
+            if (draftLine) {
+              return mergeLines(serverLine, draftLine);
+            }
+
+            return createLine(serverLine, draft, dispatch);
+          });
```

Comment:
Now, after fetching server data, iterate over and if there exists a version of that line in the draft, do a 'merge' - this was primarily done now to make the `isCreated` set to `false` after it had been created.

I don't know if we'd ever be able to set `isUpdated` to false without blocking while saving (i.e. a row `isUpdated=true` -> save -> edit same row again -> save success, don't know what to mark as updated) or having another state field 🤔  - was thinking to just send all lines as updates for now... however, without knowing this we can't determine if the invoice has been fully persisted and show some sort of warning when leaving the page if not.

There's also a chance here:
1. Add a bunch of lines
2. Save
3. Add a new line
4. when merging, only takes the server lines, so newly added line is deleted

Think you can just append any lines in the draft which aren't in the server data, though